// @d2api/manifest/generated-funcs.js
import { getDef, getAllDefs } from "./index.js";
export function getActivityDef(hash) {
    return getDef("Activity", hash);
}
export function getAllActivityDefs() {
    return getAllDefs("Activity");
}
export function getActivityGraphDef(hash) {
    return getDef("ActivityGraph", hash);
}
export function getAllActivityGraphDefs() {
    return getAllDefs("ActivityGraph");
}
export function getActivityModeDef(hash) {
    return getDef("ActivityMode", hash);
}
export function getAllActivityModeDefs() {
    return getAllDefs("ActivityMode");
}
export function getActivityModifierDef(hash) {
    return getDef("ActivityModifier", hash);
}
export function getAllActivityModifierDefs() {
    return getAllDefs("ActivityModifier");
}
export function getActivityTypeDef(hash) {
    return getDef("ActivityType", hash);
}
export function getAllActivityTypeDefs() {
    return getAllDefs("ActivityType");
}
export function getArtifactDef(hash) {
    return getDef("Artifact", hash);
}
export function getAllArtifactDefs() {
    return getAllDefs("Artifact");
}
export function getBreakerTypeDef(hash) {
    return getDef("BreakerType", hash);
}
export function getAllBreakerTypeDefs() {
    return getAllDefs("BreakerType");
}
export function getChecklistDef(hash) {
    return getDef("Checklist", hash);
}
export function getAllChecklistDefs() {
    return getAllDefs("Checklist");
}
export function getClassDef(hash) {
    return getDef("Class", hash);
}
export function getAllClassDefs() {
    return getAllDefs("Class");
}
export function getCollectibleDef(hash) {
    return getDef("Collectible", hash);
}
export function getAllCollectibleDefs() {
    return getAllDefs("Collectible");
}
export function getDamageTypeDef(hash) {
    return getDef("DamageType", hash);
}
export function getAllDamageTypeDefs() {
    return getAllDefs("DamageType");
}
export function getDestinationDef(hash) {
    return getDef("Destination", hash);
}
export function getAllDestinationDefs() {
    return getAllDefs("Destination");
}
export function getEnergyTypeDef(hash) {
    return getDef("EnergyType", hash);
}
export function getAllEnergyTypeDefs() {
    return getAllDefs("EnergyType");
}
export function getEquipmentSlotDef(hash) {
    return getDef("EquipmentSlot", hash);
}
export function getAllEquipmentSlotDefs() {
    return getAllDefs("EquipmentSlot");
}
export function getEventCardDef(hash) {
    return getDef("EventCard", hash);
}
export function getAllEventCardDefs() {
    return getAllDefs("EventCard");
}
export function getFactionDef(hash) {
    return getDef("Faction", hash);
}
export function getAllFactionDefs() {
    return getAllDefs("Faction");
}
export function getGenderDef(hash) {
    return getDef("Gender", hash);
}
export function getAllGenderDefs() {
    return getAllDefs("Gender");
}
export function getGuardianRankConstantsDef(hash) {
    return getDef("GuardianRankConstants", hash);
}
export function getAllGuardianRankConstantsDefs() {
    return getAllDefs("GuardianRankConstants");
}
export function getGuardianRankDef(hash) {
    return getDef("GuardianRank", hash);
}
export function getAllGuardianRankDefs() {
    return getAllDefs("GuardianRank");
}
export function getInventoryBucketDef(hash) {
    return getDef("InventoryBucket", hash);
}
export function getAllInventoryBucketDefs() {
    return getAllDefs("InventoryBucket");
}
export function getInventoryItemDef(hash) {
    return getDef("InventoryItem", hash);
}
export function getAllInventoryItemDefs() {
    return getAllDefs("InventoryItem");
}
export function getInventoryItemLiteDef(hash) {
    return getDef("InventoryItemLite", hash);
}
export function getAllInventoryItemLiteDefs() {
    return getAllDefs("InventoryItemLite");
}
export function getItemCategoryDef(hash) {
    return getDef("ItemCategory", hash);
}
export function getAllItemCategoryDefs() {
    return getAllDefs("ItemCategory");
}
export function getItemTierTypeDef(hash) {
    return getDef("ItemTierType", hash);
}
export function getAllItemTierTypeDefs() {
    return getAllDefs("ItemTierType");
}
export function getLoadoutColorDef(hash) {
    return getDef("LoadoutColor", hash);
}
export function getAllLoadoutColorDefs() {
    return getAllDefs("LoadoutColor");
}
export function getLoadoutConstantsDef(hash) {
    return getDef("LoadoutConstants", hash);
}
export function getAllLoadoutConstantsDefs() {
    return getAllDefs("LoadoutConstants");
}
export function getLoadoutIconDef(hash) {
    return getDef("LoadoutIcon", hash);
}
export function getAllLoadoutIconDefs() {
    return getAllDefs("LoadoutIcon");
}
export function getLoadoutNameDef(hash) {
    return getDef("LoadoutName", hash);
}
export function getAllLoadoutNameDefs() {
    return getAllDefs("LoadoutName");
}
export function getLocationDef(hash) {
    return getDef("Location", hash);
}
export function getAllLocationDefs() {
    return getAllDefs("Location");
}
export function getLoreDef(hash) {
    return getDef("Lore", hash);
}
export function getAllLoreDefs() {
    return getAllDefs("Lore");
}
export function getMaterialRequirementSetDef(hash) {
    return getDef("MaterialRequirementSet", hash);
}
export function getAllMaterialRequirementSetDefs() {
    return getAllDefs("MaterialRequirementSet");
}
export function getMedalTierDef(hash) {
    return getDef("MedalTier", hash);
}
export function getAllMedalTierDefs() {
    return getAllDefs("MedalTier");
}
export function getMetricDef(hash) {
    return getDef("Metric", hash);
}
export function getAllMetricDefs() {
    return getAllDefs("Metric");
}
export function getMilestoneDef(hash) {
    return getDef("Milestone", hash);
}
export function getAllMilestoneDefs() {
    return getAllDefs("Milestone");
}
export function getObjectiveDef(hash) {
    return getDef("Objective", hash);
}
export function getAllObjectiveDefs() {
    return getAllDefs("Objective");
}
export function getPlaceDef(hash) {
    return getDef("Place", hash);
}
export function getAllPlaceDefs() {
    return getAllDefs("Place");
}
export function getPlugSetDef(hash) {
    return getDef("PlugSet", hash);
}
export function getAllPlugSetDefs() {
    return getAllDefs("PlugSet");
}
export function getPowerCapDef(hash) {
    return getDef("PowerCap", hash);
}
export function getAllPowerCapDefs() {
    return getAllDefs("PowerCap");
}
export function getPresentationNodeDef(hash) {
    return getDef("PresentationNode", hash);
}
export function getAllPresentationNodeDefs() {
    return getAllDefs("PresentationNode");
}
export function getProgressionDef(hash) {
    return getDef("Progression", hash);
}
export function getAllProgressionDefs() {
    return getAllDefs("Progression");
}
export function getProgressionLevelRequirementDef(hash) {
    return getDef("ProgressionLevelRequirement", hash);
}
export function getAllProgressionLevelRequirementDefs() {
    return getAllDefs("ProgressionLevelRequirement");
}
export function getProgressionMappingDef(hash) {
    return getDef("ProgressionMapping", hash);
}
export function getAllProgressionMappingDefs() {
    return getAllDefs("ProgressionMapping");
}
export function getRaceDef(hash) {
    return getDef("Race", hash);
}
export function getAllRaceDefs() {
    return getAllDefs("Race");
}
export function getRecordDef(hash) {
    return getDef("Record", hash);
}
export function getAllRecordDefs() {
    return getAllDefs("Record");
}
export function getReportReasonCategoryDef(hash) {
    return getDef("ReportReasonCategory", hash);
}
export function getAllReportReasonCategoryDefs() {
    return getAllDefs("ReportReasonCategory");
}
export function getRewardSourceDef(hash) {
    return getDef("RewardSource", hash);
}
export function getAllRewardSourceDefs() {
    return getAllDefs("RewardSource");
}
export function getSandboxPatternDef(hash) {
    return getDef("SandboxPattern", hash);
}
export function getAllSandboxPatternDefs() {
    return getAllDefs("SandboxPattern");
}
export function getSandboxPerkDef(hash) {
    return getDef("SandboxPerk", hash);
}
export function getAllSandboxPerkDefs() {
    return getAllDefs("SandboxPerk");
}
export function getSeasonDef(hash) {
    return getDef("Season", hash);
}
export function getAllSeasonDefs() {
    return getAllDefs("Season");
}
export function getSeasonPassDef(hash) {
    return getDef("SeasonPass", hash);
}
export function getAllSeasonPassDefs() {
    return getAllDefs("SeasonPass");
}
export function getSocialCommendationDef(hash) {
    return getDef("SocialCommendation", hash);
}
export function getAllSocialCommendationDefs() {
    return getAllDefs("SocialCommendation");
}
export function getSocialCommendationNodeDef(hash) {
    return getDef("SocialCommendationNode", hash);
}
export function getAllSocialCommendationNodeDefs() {
    return getAllDefs("SocialCommendationNode");
}
export function getSocketCategoryDef(hash) {
    return getDef("SocketCategory", hash);
}
export function getAllSocketCategoryDefs() {
    return getAllDefs("SocketCategory");
}
export function getSocketTypeDef(hash) {
    return getDef("SocketType", hash);
}
export function getAllSocketTypeDefs() {
    return getAllDefs("SocketType");
}
export function getStatDef(hash) {
    return getDef("Stat", hash);
}
export function getAllStatDefs() {
    return getAllDefs("Stat");
}
export function getStatGroupDef(hash) {
    return getDef("StatGroup", hash);
}
export function getAllStatGroupDefs() {
    return getAllDefs("StatGroup");
}
export function getTalentGridDef(hash) {
    return getDef("TalentGrid", hash);
}
export function getAllTalentGridDefs() {
    return getAllDefs("TalentGrid");
}
export function getTraitDef(hash) {
    return getDef("Trait", hash);
}
export function getAllTraitDefs() {
    return getAllDefs("Trait");
}
export function getUnlockDef(hash) {
    return getDef("Unlock", hash);
}
export function getAllUnlockDefs() {
    return getAllDefs("Unlock");
}
export function getUnlockValueDef(hash) {
    return getDef("UnlockValue", hash);
}
export function getAllUnlockValueDefs() {
    return getAllDefs("UnlockValue");
}
export function getVendorDef(hash) {
    return getDef("Vendor", hash);
}
export function getAllVendorDefs() {
    return getAllDefs("Vendor");
}
export function getVendorGroupDef(hash) {
    return getDef("VendorGroup", hash);
}
export function getAllVendorGroupDefs() {
    return getAllDefs("VendorGroup");
}
